spec/integration/api/search_controller_spec.rb

```diff
@@ -0,0 +1,155 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe Api::V1::SearchController do
+  let(:auth) { FactoryGirl.create(:auth_with_read_and_write) }
+  let!(:access_token) { auth.create_access_token.to_s }
+  let(:auth_read_only) { FactoryGirl.create(:auth_with_read) }
+  let!(:access_token_read_only) { auth_read_only.create_access_token.to_s }
+
+  describe "#user_index" do
+    before do
+      @searchable_user = FactoryGirl.create(
+        :person,
+        diaspora_handle: "findable@example.org",
+        profile:         FactoryGirl.build(:profile, first_name: "Terry", last_name: "Smith")
+      )
+
+      @closed_user = FactoryGirl.create(
+        :person,
+        closed_account: true,
+        profile:        FactoryGirl.build(:profile, first_name: "Closed", last_name: "Account")
+      )
+      @unsearchable_user = FactoryGirl.create(
+        :person,
+        diaspora_handle: "unsearchable@example.org",
+        profile:         FactoryGirl.build(
+          :profile,
+          first_name: "Unsearchable",
+          last_name:  "Person",
+          searchable: false
+        )
+      )
+    end
+
+    it "succeeds by tag" do
+      get(
+        "/api/v1/search/users",
+        params: {tag: "one", access_token: access_token}
+      )
+      expect(response.status).to eq(200)
+      users = JSON.parse(response.body)
+      expect(users.length).to eq(14)
+    end
+
+    it "succeeds by name" do
+      get(
+        "/api/v1/search/users",
+        params: {name_or_handle: "Terry", access_token: access_token}
+      )
+      expect(response.status).to eq(200)
+      users = JSON.parse(response.body)
+      expect(users.length).to eq(1)
+    end
+
+    it "succeeds by handle" do
+      get(
+        "/api/v1/search/users",
+        params: {name_or_handle: "findable", access_token: access_token}
+      )
+      expect(response.status).to eq(200)
+      users = JSON.parse(response.body)
+      expect(users.length).to eq(1)
+    end
+
+    it "doesn't return closed accounts" do
+      get(
+        "/api/v1/search/users",
+        params: {name_or_handle: "Closed", access_token: access_token}
+      )
+      expect(response.status).to eq(200)
+      users = JSON.parse(response.body)
+      expect(users.length).to eq(0)
+    end
+
+    it "doesn't return hidden accounts" do
+      get(
+        "/api/v1/search/users",
+        params: {name_or_handle: "Unsearchable", access_token: access_token}
```

Comment:
I don't know if search is case sensitive. But if it's the case, it should be `unsearchable` here.

Reply:
I was torn on weather to use the diaspora handle name or the first name.  I'll make it the handle so that it is clear that it is returning nothing because of the unsearchable flag rather than the fact the query itself isn't correct.

Reply:
Changed to use the diaspora handle explicitly